src/utils/with-react.js

```diff
@@ -1,3 +1,75 @@
 import val from '@skatejs/val';
 
+// defined values are different from undefined (for properties) or null (for attributes)
+const isDefined = value => !(value === undefined || value === null);
+
+// generic defaults (will be applied in the absence of explicit 'defaultValue')
+const DEFAULT_VALUE_OF_TYPE = new Map([
+  [String, ''],
+  [Boolean, false],
+  [Object, {}],
+  [Array, []],
+  [Number, 0],
+  [Function, () => {}],
+]);
+
+const convert = (value, type) => {
+  if (type === Number) {
+    return parseFloat(value);
+  }
+  return type === Array || type === Object ? JSON.parse(value) : value;
+};
+
+const applyDefaults = ceInst => {
+  const {
+    constructor: { properties },
+  } = ceInst;
+  // get all properties of the custom element and loop over each key
+  Object.keys(properties).forEach(property => {
+    // extract default value and property type found in the name of the Type
```

Comment:
type is uppercase, plz correct to lowercase

Reply:
fixed